python/cudf/cudf/core/column/timedelta.py

```diff
@@ -0,0 +1,402 @@
+                    ),
+                ),
+                "hours": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["D"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["h"], "ns")
+                    ),
+                ),
+                "minutes": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["h"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["m"], "ns")
+                    ),
+                ),
+                "seconds": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["m"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["s"], "ns")
+                    ),
+                ),
+                "milliseconds": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["s"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["ms"], "ns")
+                    ),
+                ),
+                "microseconds": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["ms"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["us"], "ns")
+                    ),
+                ),
+                "nanoseconds": self.binary_operator(
+                    "mod",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["us"], "ns")
+                    ),
+                ).binary_operator(
+                    "floordiv",
+                    as_scalar(
+                        np.timedelta64(_numpy_to_pandas_conversion["ns"], "ns")
+                    ),
+                ),
+            },
+            index=index,
+        )
+
+    @property
+    def days(self):
+        return self.binary_operator(
+            "floordiv",
+            as_scalar(np.timedelta64(_numpy_to_pandas_conversion["D"], "ns")),
+        )
```

Comment:
So this is because of the reason we support different dtypes in our timedelta column as opposed to pandas only supporting `timedelta64[ns]`. 
Incase of pandas Series(`timedelta64[ns]`) if we divide the series by `1 Day`(np.timedelta64(1, 'D')) or `1 second`(np.timedelta64(1, 's')) will result in number of `Days` or `Seconds` respectively. 
Due to the extra dtypes support we have, instead of type-casting `self` to `ns` and performing the division with `1 Day` or `1 second`, we can create an equivalent `ns` denominator and perform a `floordiv`. This still will go down `floordiv` binop flow and type-cast to a common unit(which is `ns`) in this case to yield the number of days or seconds respectively..